lib/edge-runtime/requests.ts

```diff
@@ -60,70 +63,76 @@ export function constructHttpRequest({
-      throw new Error(
-        "Only application/json request body contents are supported"
-      );
-    }
-    const applicationJson = reqBodyContents[
-      "application/json"
-    ] as OpenAPIV3.MediaTypeObject;
-
-    const schema = applicationJson.schema as OpenAPIV3.SchemaObject;
-    console.log("schema:", JSON.stringify(schema));
-    const properties = schema.properties as {
-      [name: string]: OpenAPIV3_1.SchemaObject;
-    };
+    const { properties, schema } = bodyPropertiesFromRequestBodyContents(
+      action.request_body_contents
+    );
     console.log("properties:", JSON.stringify(properties));
-    const required = schema.required ?? [];
-    const bodyArray = Object.entries(properties).map(([name, property]) => {
-      // Throw out readonly attributes
-      if (property.readOnly) return undefined;
-      if (parameters[name]) {
-        return { [name]: parameters[name] };
-      } else if (
-        // If the parameter is a required enum with 1 value
-        property.enum &&
-        property.enum.length === 1 &&
-        required.includes(name)
-      ) {
-        return { [name]: property.enum[0] };
-      }
-    });
-    console.log("bodyArray:", JSON.stringify(bodyArray));
+    const body = buildBody(schema, properties, parameters);
+    requestOptions.body = JSON.stringify(body);
+  }
 
-    const body = Object.assign({}, ...bodyArray);
+  let url = buildUrl(organization, action, parameters, headers);
+  const logMessage = `Attempting fetch with url: ${url}\n\nWith options:${JSON.stringify(
+    requestOptions,
+    null,
+    2
+  )}`;
+  console.log(logMessage);
 
-    // Check all required params are present
-    required.forEach((key: string) => {
-      // TODO: This doesn't check nested required fields
-      if (!body[key]) {
-        throw new Error(
-          `Required parameter "${key}" not provided to action: ${
-            action.name
-          } out of ${JSON.stringify(Object.keys(body))}`
-        );
-      }
+  if (stream)
+    stream({
+      role: "debug",
+      content: logMessage,
     });
-    requestOptions.body = JSON.stringify(body);
+
+  return { url, requestOptions };
+}
+
+export function bodyPropertiesFromRequestBodyContents(
+  requestBodyContents: Json
+) {
+  const reqBodyContents =
+    requestBodyContents as unknown as OpenAPIV3.RequestBodyObject;
+
+  console.log("reqBodyContents:", reqBodyContents);
+
+  // TODO: Only application/json supported for now
+  if (!("application/json" in reqBodyContents)) {
+    throw new Error(
+      "Only application/json request body contents are supported"
+    );
   }
+  const applicationJson = reqBodyContents[
+    "application/json"
+  ] as OpenAPIV3.MediaTypeObject;
+
+  const schema = applicationJson.schema as OpenAPIV3.SchemaObject;
+  const properties = schema.properties as {
+    [name: string]: OpenAPIV3_1.SchemaObject;
+  };
+  return { properties, schema };
+}
 
+function buildUrl(
```

Comment:
I find this function confusing because it's called `buildUrl()` and are just returning the url, but we're making changes to the `headers` variable in this function :confused:

Reply:
This is a very good spot, but I think this is an issue in main too? Changes are made to the headers variable that are never used anywhere right? or are the changes made to a reference to requestOptions.headers (in which case this is confusing imo)

Reply:
Yep, think you're right that they _should_ say `requestOptions.headers`, although by luck, it didn't break anything because changes to the `headers` object were also applied to `requestOptions.headers`